File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/execution/IncludedSignatureWithAliasOutputTests.java
package org.opencds.cqf.cql.engine.execution;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import java.math.BigDecimal;
import org.testng.annotations.Test;

public class IncludedSignatureWithAliasOutputTests extends CqlTestBase {

    @Test
    public void testEvaluate() {
        var results = engine.evaluate(toElmIdentifier("IncludedSignatureWithAliasOutputTests"));

        Object value = results.forExpression("TestOverload").value();
        assertThat(value, is(5));

        value = results.forExpression("TestOverloadOneInt").value();
        assertThat(value, is(1));

        value = results.forExpression("TestOverloadOneDecimal").value();
        assertThat(value, is(new BigDecimal("2.0")));

        value = results.forExpression("TestOverloadTwoInts").value();
        assertThat(value, is(1));

        value = results.forExpression("TestOverloadTwoDecimals").value();
        assertThat(value, is(new BigDecimal("2.0")));

        value = results.forExpression("TestOverloadOneIntOneDecimal").value();
        assertThat(value, is(new BigDecimal("2.0")));

        value = results.forExpression("TestOverloadOneIntTwoDecimal").value();
        assertThat(value, is(1));
    }
}

File: Src/java/engine/src/test/java/org/opencds/cqf/cql/engine/execution/IncludedSignatureWithAliasOutputTests.kt
package org.opencds.cqf.cql.engine.execution

import java.math.BigDecimal
import org.hamcrest.MatcherAssert
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test

internal class IncludedSignatureWithAliasOutputTests : CqlTestBase() {
    @Test
    fun evaluate() {
        val results = engine.evaluate(toElmIdentifier("IncludedSignatureWithAliasOutputTests"))

        var value = results.forExpression("TestOverload").value()
        MatcherAssert.assertThat(value, Matchers.`is`(5))

        value = results.forExpression("TestOverloadOneInt").value()
        MatcherAssert.assertThat(value, Matchers.`is`(1))

        value = results.forExpression("TestOverloadOneDecimal").value()
        MatcherAssert.assertThat(value, Matchers.`is`(BigDecimal("2.0")))

        value = results.forExpression("TestOverloadTwoInts").value()
        MatcherAssert.assertThat(value, Matchers.`is`(1))

        value = results.forExpression("TestOverloadTwoDecimals").value()
        MatcherAssert.assertThat(value, Matchers.`is`(BigDecimal("2.0")))

        value = results.forExpression("TestOverloadOneIntOneDecimal").value()
        MatcherAssert.assertThat(value, Matchers.`is`(BigDecimal("2.0")))

        value = results.forExpression("TestOverloadOneIntTwoDecimal").value()
        MatcherAssert.assertThat(value, Matchers.`is`(1))
    }
}
